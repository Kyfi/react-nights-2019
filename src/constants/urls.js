const url = {
  index: '/',
  signUp: '/sign_up',
  signIn: '/sign_in',
  cart: '/cart',
  productDetail: productId => `/${productId}`,
}

export default url
