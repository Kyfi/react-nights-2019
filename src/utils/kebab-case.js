export const kebabCase = ( value ) => {
  const kebabed = value.split(/[_\s]/).join("-")
  return kebabed
}
