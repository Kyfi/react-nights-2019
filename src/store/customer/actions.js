import { getCustomerToken } from '../../api/customers/getCustomerToken'
import { getCustomer } from '../../api/customers/getCustomer'
import urls from '../../constants/urls'

import { setToken, removeToken } from '../../utils/token'
import { setRefreshToken, removeRefreshToken } from '../../utils/refreshToken'
import { removeCustomer } from '../../utils/customer'

export const LOGIN_INIT = 'customer/LOGIN_INIT'
export const LOGIN_FAIL = 'customer/LOGIN_FAIL'
export const LOGIN_SUCCESS = 'customer/LOGIN_SUCCESS'

export const LOGOUT = 'customer/LOGOUT'

export const login = ({ username, password, push }) => async dispatch => {
  dispatch({
    type: LOGIN_INIT,
    payload: { username, password },
  })

  const { ownerId, access_token, refresh_token } = await getCustomerToken({
    username,
    password,
  })

  setToken(access_token)
  setRefreshToken(refresh_token)

  const customer = await getCustomer(ownerId)

  dispatch({
    type: LOGIN_SUCCESS,
    payload: {
      customer,
    },
  })

  push(urls.myAccount)
}

export const logout = ({ push }) => dispatch => {
  removeToken()
  removeRefreshToken()
  removeCustomer()

  push(urls.index)

  dispatch({
    type: LOGOUT,
  })
}
