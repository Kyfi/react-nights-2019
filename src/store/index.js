import { createStore, combineReducers } from 'redux'

import cart from './cart/reducer'
import customer from './customer/reducer'

const reducer = combineReducers({
  cart,
  customer,
})

export const configureStore = (preloadedState = {}) =>
  createStore(
    reducer,
    preloadedState,
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
  )
