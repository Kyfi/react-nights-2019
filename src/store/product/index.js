import { LOAD_PRODUCT } from './actions'

const reducer = (state = null, action) => {
  switch (action.type) {
    case LOAD_PRODUCT:
      return action.payload
    default:
      return state
  }
}

export default reducer
