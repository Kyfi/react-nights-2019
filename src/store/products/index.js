import { LOAD_PRODUCTS, CHANGE_PAGE_SIZE } from './actions'

const reducer = (state = [], action) => {
  switch (action.type) {
    case LOAD_PRODUCTS:
      return action.payload
    case CHANGE_PAGE_SIZE:
      return action.payload
    default:
      return state
  }
}

export default reducer
