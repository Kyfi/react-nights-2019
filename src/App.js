import React from 'react'
import { Provider } from 'react-redux'
import { ToastContainer, toast } from 'react-toastify'

import Routes from './routes'
import GlobalStyles from './globalStyles'
import { getCustomer } from './utils/customer'
import { configureStore } from './store'
import { ErrorBoundary } from './components/ErrorBoundary'

const defaultStore = configureStore({
  customer: getCustomer(),
})

const App = ({ store }) => (
  <Provider store={store || defaultStore}>
    <React.Fragment>
      <GlobalStyles />
      <ToastContainer position={toast.POSITION.BOTTOM_RIGHT} />
      <ErrorBoundary>
        <Routes />
      </ErrorBoundary>
    </React.Fragment>
  </Provider>
)

export { App }
