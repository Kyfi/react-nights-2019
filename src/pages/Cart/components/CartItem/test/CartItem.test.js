import React from 'react'
import { waitForElement } from 'react-testing-library'
import 'jest-styled-components'

import { App } from '../../../../../App'
import urls from '../../../../../constants/urls'
import { renderWithRouter } from '../../../../../utilsTest/render'
import { mockFetchProduct } from '../../../../../utilsTest/mockHelpers'
import { configureStore } from '../../../../../store'
import { getCustomer } from '../../../../../utils/customer'

describe('[pages] Cart', () => {
  describe('when loading', () => {
    it('should render correctly', () => {
      const renderer = renderWithRouter(<App />, urls.cart)
      expect(renderer.container).toMatchSnapshot()
    })
  })

  describe('when products loaded', () => {
    mockFetchProduct()

    it('should render correctly', async () => {
      const productId = 1
      const store = configureStore({
        customer: getCustomer(),
        cart: {
          [productId]: productId,
        },
      })

      const renderer = renderWithRouter(<App store={store} />, urls.cart)
      await waitForElement(() => renderer.getByTestId('product-in-cart'))
      expect(renderer.getAllByTestId('product-in-cart').length).toEqual(1)
    })
  })
})
