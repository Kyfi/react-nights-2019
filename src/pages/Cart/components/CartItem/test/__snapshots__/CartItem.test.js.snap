// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[pages] Cart when loading should render correctly 1`] = `
.c5 {
  padding: 2rem;
}

.c0 {
  position: relative;
  z-index: 1;
}

.c1 {
  background: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  box-shadow: 0 0 5rem -1.3rem rgb(150,150,150);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #252522;
  background-color: #fff;
  padding: 20px;
  border: 0;
  border-right: 1px solid lightgrey;
  white-space: nowrap;
  font-weight: 300;
}

.c3:hover {
  color: #6364d9;
}

.c6 {
  font-weight: 400;
  font-family: circular,serif;
  color: #6364d9;
  text-transform: uppercase;
}

<div>
  <div
    class="Toastify"
  />
  <div
    class="c0"
  >
    <nav
      class="c1"
    >
      <div
        class="c2"
      >
        <a
          class="c3"
          href="/"
        >
          Product List
        </a>
      </div>
      <div
        class="c4"
      >
        <a
          class="c3"
          href="/sign_in"
        >
          Sign In
        </a>
        <a
          class="c3"
          href="/sign_up"
        >
          Sign Up
        </a>
        <a
          class="c3"
          href="/cart"
        >
          My Cart
        </a>
      </div>
    </nav>
  </div>
  <div
    class="c5"
  >
    <h1
      class="c6"
    >
      Cart
    </h1>
    <div>
      <span
        aria-label="Cart is so empty"
        role="img"
      >
        ðŸ˜¢
      </span>
       
      Cart is so empty...
    </div>
  </div>
</div>
`;
